usaspending_api/common/serializers.py

```diff
@@ -7,83 +7,127 @@ class LimitableSerializer(serializers.ModelSerializer):
-            if exclude_fields is not None and not self.context.get("child", False):  # the child check should be removed when child field limiting is implemented
-                for field_name in exclude_fields:
-                    try:
-                        self.fields.pop(field_name)
-                    except KeyError:
-                        # Because we're not currently handling nested serializer field
-                        # limiting pass-down, this can happen due to the context pass down
-                        pass
-
-            if include_fields is not None and not self.context.get("child", False):  # the child check should be removed when child field limiting is implemented
-                allowed = set(include_fields)
-                existing = set(self.fields.keys())
-                for field_name in existing - allowed:
-                    try:
-                        # If we have a coded field, always include its description
-                        if field_name.split("_")[-1] == "description" and "_".join(field_name.split("_")[:-1]) not in allowed:
-                            continue
-                        self.fields.pop(field_name)
-                    except KeyError:
-                        # Because we're not currently handling nested serializer field
-                        # limiting pass-down, this can happen due to the context pass down
-                        pass
-
-            elif current_viewset and current_viewset.action == 'retrieve':
-                # The view has specifically asked that all fields should
-                # be returned (for example, when the request url follows
-                # the item/{pk} pattern)
-                return
-
-            else:
-                try:
-                    include_fields = self.Meta.model.get_default_fields(path=request._request.path_info)
-                    allowed = set(include_fields)
-                    existing = set(self.fields.keys())
-                    for field_name in existing - allowed:
-                        self.fields.pop(field_name)
-                except AttributeError:
-                    # We don't have get default fields available
-                    pass
+        # Now we alter our own field sets
+        # We must exclude before include to avoid conflicts from user error
+        for field_name in exclude_fields:
+            self.fields.pop(field_name)
+
+        if len(include_fields) > 0:
+            allowed = set(include_fields)
+            existing = set(self.fields.keys())
+            for field_name in existing - allowed:
+                # If we have a coded field, always include its description
+                if field_name.split("_")[-1] == "description" and "_".join(field_name.split("_")[:-1]) not in allowed:
+                    continue
+                self.fields.pop(field_name)
+
+    # Returns a list of child names for fields in the field list
```

Comment:
Thanks, as always, for taking the time to document. Now that there's a function for this logic, what do you think about converting these remarks to the function's docstring?